builder/amazon/common/step_run_spot_instance.go

```diff
@@ -260,17 +260,30 @@ func (s *StepRunSpotInstance) Run(ctx context.Context, state multistep.StateBag)
 				Tags:         spotTags,
 			},
 		}
+		launchTemplate.LaunchTemplateData.TagSpecifications = []*ec2.LaunchTemplateTagSpecificationRequest{
+			{
+				ResourceType: aws.String("instance"),
+				Tags:         spotTags,
```

Comment:
We have a large and confusing number of tags in the amazon builder; I think this should be set using the run_tags, rather than the spot_tags

Reply:
Yep, makes sense.